prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1688,6 +1688,8 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 			ret = rxr_pkt_post_data(ep, tx_entry);
 			if (OFI_UNLIKELY(ret)) {
 				tx_entry->send_flags &= ~FI_MORE;
+				if (ret == -FI_EAGAIN)
+					goto out;
```

Comment:
With this change, do we still need the previous block
``` 
if (ep->tx_pending == ep->max_outstanding_tx)
	goto out;
```?

Reply:
Yes, in case for whatever reason the tx pool becomes larger than the tx queue depth.

Reply:
Thanks!